mathesar/api/exceptions/exceptions.py

```diff
@@ -0,0 +1,46 @@
+from collections import namedtuple
+
+from django.utils.encoding import force_str
+from rest_framework import status
+from rest_framework.exceptions import APIException as DRFApiException
+
+from mathesar.api.exceptions.error_codes import ErrorCodes
+
+ErrorBody = namedtuple(
+    'ErrorBody',
+    ['code', 'message', 'field', 'details'],
+    defaults=[None, None]
+)
+
+
+def get_default_exception_detail(exception, error_code=ErrorCodes.NonClassifiedError.value,
+                                 message=None, field=None, details=None):
+    return ErrorBody(
+        message=force_str(exception) if message is None else message,
+        code=error_code,
+        field=field,
+        details=details
+    )
+
+
+def get_default_api_exception(exc):
+    return APIException(exc, ErrorCodes.NonClassifiedError.value)
+
+
+class APIException(DRFApiException):
```

Comment:
Can we call ours `MathesarAPIException`? It might be confusing to overload the original name for contributors who are used to working with DRF.